source/s3_request_messages.c

```diff
@@ -608,13 +668,50 @@ struct aws_input_stream *aws_s3_message_util_assign_body(
         goto error_clean_up;
     }
 
+    bool is_get = false;
+    if (s_is_get_request(out_message, &is_get)) {
+        goto error_clean_up;
+    }
+    if (is_get) {
+        aws_http_headers_set(headers, g_request_validation_mode, g_enabled);
+    }
+
+    if (algorithm) {
+        /* set Content-Encoding header */
+        if (!is_get && aws_http_headers_set(headers, g_content_encoding_header_name, g_content_encoding_header_value)) {
```

Comment:
`g_content_encoding_header_value = "aws-chunked"`
Rename it, there are multiple encoding types, this is a specific one so naming it after the `aws-chunked`.